tooling/cli/src/interface/rust.rs

```diff
@@ -217,8 +217,24 @@ impl Interface for Rust {
 
     let mut s = self.app_settings.target_triple.split('-');
     let (arch, _, host) = (s.next().unwrap(), s.next().unwrap(), s.next().unwrap());
-    env.insert("TAURI_ARCH", arch.into());
-    env.insert("TAURI_PLATFORM", host.into());
+    env.insert(
+      "TAURI_ARCH",
+      match arch {
+        // keeps compatibility with old `std::env::consts::ARCH` implementation
+        "i686" | "i586" => "x86".into(),
+        a => a.into(),
+      },
+    );
+    env.insert(
+      "TAURI_PLATFORM",
+      match host {
+        // keeps compatibility with old `std::env::consts::OS` implementation
+        "darwin" => "macos".into(),
+        "ios-sim" => "ios".into(),
+        "androideabi" => "android".into(),
+        h => h.into(),
+      },
+    );
```

Comment:
but the old implementation was just wrong and I expect `TAURI_ARCH` will be used in build tools so for example `darwin`, `androideabi` will be the correct value.

Reply:
they can use the target triple for that. we can't break this, someone is probably using this (specially the platform one).

Reply:
This would make sense for the platform but for the arch, I don't like it tbh and I think users should be aware that they are using the wrong arch.

Reply:
tbh I see this as a bugfix but lets add TODO for v2 then

Reply:
what's the difference between i686 and x86? it's the same information, and x86 is the format we were using already. I think you're mixing the arch and the platform.

Reply:
one could be trying to download a sidecar (built with rust) at build time using the arch, and maybe they ended up converting `x86` to `i686` or `i586` (if someone still using that) but `x86` wasn't our intended result in the first place so this should be considered a fix because we are delivering the intended behavior even if it breaks their build script.

Reply:
they probably are but they can't map it back to i586 for example and they would need to use the target triple but that's beside the point. I simply don't want to wait until v2 to change this wrong behavior and maybe more people will end up depending on this wrong behavior. Also we are changing the tauri-cli which is just a tool and we are not breaking the framework.